eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -505,16 +505,15 @@ final case class DATA_WAIT_FOR_DUAL_FUNDING_CREATED(channelId: ByteVector32,
                                                     txBuilder: typed.ActorRef[InteractiveTxBuilder.Command],
                                                     deferred: Option[ChannelReady]) extends TransientChannelData
 final case class DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED(metaCommitments: MetaCommitments,
-                                                      fundingParams: InteractiveTxParams,
                                                       localPushAmount: MilliSatoshi,
                                                       remotePushAmount: MilliSatoshi,
                                                       waitingSince: BlockHeight, // how long have we been waiting for a funding tx to confirm
                                                       lastChecked: BlockHeight, // last time we checked if the channel was double-spent
                                                       rbfStatus: RbfStatus,
                                                       deferred: Option[ChannelReady]) extends PersistentChannelData {
-  def latestFundingTx: SignedSharedTransaction = metaCommitments.latest.localFundingStatus.asInstanceOf[DualFundedUnconfirmedFundingTx].sharedTx
-  def previousFundingTxs: Seq[SignedSharedTransaction] = metaCommitments.commitments.map(_.localFundingStatus).collect { case DualFundedUnconfirmedFundingTx(sharedTx, _) if sharedTx.txId != latestFundingTx.txId => sharedTx }
-  def allFundingTxs: Seq[SignedSharedTransaction] = latestFundingTx +: previousFundingTxs
+  def latestFundingTx: DualFundedUnconfirmedFundingTx = allFundingTxs.head
```

Comment:
Using `.head` makes an assumption on the ordering used inside `metaCommitments.commitments`, that's why I'd rather access it more explicitly via `metaCommitments.latest` which removes that assumption, but if you think it's acceptable we can keep it that way.

Reply:
I used `.head` because I was also using `.tail`, and it allowed me to remove the cast to `DualFundedUnconfirmedFundingTx`.

Alternatively I could do:
```scala
  def allFundingTxs: Seq[DualFundedUnconfirmedFundingTx] = metaCommitments.commitments.map(_.localFundingStatus).collect { case fundingTx: DualFundedUnconfirmedFundingTx => fundingTx }
  def latestFundingTx: DualFundedUnconfirmedFundingTx = metaCommitments.latest.localFundingStatus.asInstanceOf[DualFundedUnconfirmedFundingTx]
  def previousFundingTxs: Seq[DualFundedUnconfirmedFundingTx] = allFundingTxs diff Seq(latestFundingTx)
```

Reply:
I think that's better since it decouples us from the internals of `MetaCommitments`

Reply:
I see your point: it's ugly, but morally right ^^. Done in 6ee161f69.